worker/api.go

```diff
@@ -20,6 +23,25 @@ type (
 	PrivateKey = consensus.PrivateKey
 )
 
+// A Duration is the elapsed time between two instants. Durations are encoded as
+// an integer number of milliseconds.
+type Duration time.Duration
+
+func (d Duration) MarshalText() ([]byte, error) {
+	return []byte(fmt.Sprint(time.Duration(d).Milliseconds())), nil
+}
+
+func (d *Duration) UnmarshalText(b []byte) error {
+	var ms int64
```

Comment:
Do we need to worry about this being negative?

Reply:
Added a check for it, didn't scan into uint to give a more descriptive error